banyand/index/index.go

```diff
@@ -154,6 +169,28 @@ func (s *service) GracefulStop() {
 	}
 }
 
+func (s *service) Ready(ctx context.Context, options ...ReadyOption) bool {
+	options = append(options, func(m map[string]*series) bool {
+		return len(m) > 0
+	})
+
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err() == nil
+		default:
+			allMatches := true
+			for _, opt := range options {
+				allMatches = allMatches && opt(s.meta.meta)
```

Comment:
Could you break the loop once getting a `false` result from `opt()` instead of iterating all options.

Reply:
Fixed